pkg/sync/kubernetes/featureflagconfiguration/clientset.go

```diff
@@ -0,0 +1,122 @@
+package featureflagconfiguration
+
+import (
+	"errors"
+	"time"
+
+	"github.com/open-feature/flagd/pkg/sync"
+	ffv1alpha1 "github.com/open-feature/open-feature-operator/apis/core/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type FFCInterface interface {
+	FeatureFlagConfigurations(namespace string) Interface
+}
+
+type FFCClient struct {
+	restClient rest.Interface
+}
+
+func createFuncHandler(obj interface{}, object client.ObjectKey, c chan<- sync.INotify) {
+	if obj.(*ffv1alpha1.FeatureFlagConfiguration).Name == object.Name {
+		c <- &sync.Notifier{
+			Event: sync.Event[sync.DefaultEventType]{
+				EventType: sync.DefaultEventTypeCreate,
+			},
+		}
+	}
+}
+
+func updateFuncHandler(oldObj interface{}, object client.ObjectKey, c chan<- sync.INotify) {
+	if oldObj.(*ffv1alpha1.FeatureFlagConfiguration).Name == object.Name {
+		c <- &sync.Notifier{
+			Event: sync.Event[sync.DefaultEventType]{
+				EventType: sync.DefaultEventTypeModify,
+			},
+		}
+	}
+}
+
+func deleteFuncHandler(obj interface{}, object client.ObjectKey, c chan<- sync.INotify) {
+	if obj.(*ffv1alpha1.FeatureFlagConfiguration).Name == object.Name {
+		c <- &sync.Notifier{
+			Event: sync.Event[sync.DefaultEventType]{
+				EventType: sync.DefaultEventTypeDelete,
+			},
+		}
+	}
+}
+
+func WatchResources(clientSet FFCInterface, refreshTime time.Duration,
+	object client.ObjectKey, c chan<- sync.INotify,
+) {
+	ns := "*"
+	if object.Namespace != "" {
+		ns = object.Namespace
+	}
+	_, ffConfigController := cache.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(lo metav1.ListOptions) (result runtime.Object, err error) {
+				res, err := clientSet.FeatureFlagConfigurations(ns).List(lo)
+				return res, err
+			},
+			WatchFunc: func(lo metav1.ListOptions) (watch.Interface, error) {
+				wiface, err := clientSet.FeatureFlagConfigurations(ns).Watch(lo)
```

Comment:
can return this directly rather than assigning to variables

Reply:
Yeah this was done for debugging, will revert!